consensus/hotstuff/timeoutcollector/timeout_processor_test.go

```diff
@@ -0,0 +1,502 @@
+	// prepare staking signers, each signer has its own private/public key pair
+	stakingSigners := unittest.IdentityListFixture(11, func(identity *flow.Identity) {
+		stakingPriv := unittest.StakingPrivKeyFixture()
+		identity.StakingPubKey = stakingPriv.PublicKey()
+
+		me, err := local.New(identity, stakingPriv)
+		require.NoError(t, err)
+
+		signers[identity.NodeID] = verification.NewStakingSigner(me)
+	})
+
+	// utility function which generates a valid timeout for every signer
+	createTimeouts := func(view uint64, newestQC *flow.QuorumCertificate, lastViewTC *flow.TimeoutCertificate) []*model.TimeoutObject {
+		timeouts := make([]*model.TimeoutObject, 0, len(stakingSigners))
+		for _, signer := range stakingSigners {
+			timeout, err := signers[signer.NodeID].CreateTimeout(view, newestQC, lastViewTC)
+			require.NoError(t, err)
+			timeouts = append(timeouts, timeout)
+		}
+		return timeouts
+	}
+
+	leader := stakingSigners[0]
+
+	view := uint64(rand.Uint32() + 100)
+	block := helper.MakeBlock(helper.WithBlockView(view-1),
+		helper.WithBlockProposer(leader.NodeID))
+
+	committee := mocks.NewDynamicCommittee(t)
+	committee.On("IdentitiesByEpoch", mock.Anything).Return(stakingSigners, nil)
+	committee.On("IdentityByEpoch", mock.Anything, mock.Anything).Return(leader, nil)
+	committee.On("IdentitiesByBlock", mock.Anything).Return(stakingSigners, nil)
+	committee.On("QuorumThresholdForView", mock.Anything).Return(committees.WeightThresholdToBuildQC(stakingSigners.TotalWeight()), nil)
+	committee.On("TimeoutThresholdForView", mock.Anything).Return(committees.WeightThresholdToTimeout(stakingSigners.TotalWeight()), nil)
+
+	proposal, err := signers[leader.NodeID].CreateProposal(block)
+	require.NoError(t, err)
+
+	var newestQC *flow.QuorumCertificate
+	onQCCreated := func(qc *flow.QuorumCertificate) {
+		newestQC = qc
+	}
+
+	voteProcessorFactory := votecollector.NewStakingVoteProcessorFactory(committee, onQCCreated)
+	voteProcessor, err := voteProcessorFactory.Create(unittest.Logger(), proposal)
+	require.NoError(t, err)
+
+	for _, signer := range stakingSigners[1:] {
+		vote, err := signers[signer.NodeID].CreateVote(block)
+		require.NoError(t, err)
+		err = voteProcessor.Process(vote)
+		require.NoError(t, err)
+	}
+
+	require.NotNil(t, newestQC, "vote processor must create a valid QC at this point")
+
+	// create verifier that will do crypto checks of created TC
+	verifier := verification.NewStakingVerifier()
+	// create validator which will do compliance and crypto checked of created TC
+	validator := hotstuffvalidator.New(committee, verifier)
+
+	var lastViewTC *flow.TimeoutCertificate
+	onTCCreated := func(args mock.Arguments) {
+		tc := args.Get(0).(*flow.TimeoutCertificate)
+		// check if resulted TC is valid
+		err := validator.ValidateTC(tc)
+		require.NoError(t, err)
+		lastViewTC = tc
+	}
+
+	aggregator, err := NewTimeoutSignatureAggregator(view, stakingSigners, encoding.CollectorVoteTag)
```

Comment:
Should we have a separate tag for timeout objects compared to votes?

Reply:
also great catch. Yes, we totally should.

Reply:
I've added new tags for consensus and collector timeouts